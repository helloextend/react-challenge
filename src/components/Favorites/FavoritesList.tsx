import React, { FC } from 'react'
import ImageList from '../ImageList'
import { useSelector } from 'react-redux'

const FavoriteList: FC = () => {
  const favorites = Object.keys(useSelector((state) => state.favorites))
  return <ImageList images={favorites}></ImageList>
}

export default FavoriteList
