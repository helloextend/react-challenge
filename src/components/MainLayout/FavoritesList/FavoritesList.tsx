import React, { FC } from 'react'
import styled from '@emotion/styled'
import Heart from '../../Heart';
import { colors } from '../../../assets/index';
import ResultsList from '../ResultsList/ResultsList';

interface Props {
  favorites: Array<string>,
  clickHeart: (imgUrl: string) => void,
}

const FavoritesList: FC<Props> = ({ favorites, clickHeart }) => {
  return (
    <>
      <Title><Heart style={heartStyle} icon="redHeartIcon" alt="red heart icon" />Favorites</Title>
      <ResultsList columnCount={4} imgUrls={favorites} favorites={favorites} clickHeart={clickHeart} />
    </>
  )
}

const heartStyle = {
  marginRight: '30px'
}

const Title = styled.h1({
  borderTop: `1px solid ${colors.veryLightGray}`,
  paddingTop: '30px',
  fontWeight: 'bold',
  fontSize: '24px',
  lineHeight: '33px',
})

export default FavoritesList