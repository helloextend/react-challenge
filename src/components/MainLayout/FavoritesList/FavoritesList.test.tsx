import * as React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import FavoritesList from './FavoritesList'


const baseProps = {
  favorites: [
    'https://images.dog.ceo/breeds/hound-afghan/n02088094_1003.jpg',
    'https://images.dog.ceo/breeds/hound-afghan/n02088094_1007.jpg',
    'https://images.dog.ceo/breeds/hound-afghan/n02088094_1023.jpg',
  ],
  clickHeart: jest.fn(),
};



describe('FavoritesList component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<FavoritesList {...baseProps} />)
    expect(wrapper.exists()).toBe(true)
  });

  it('renders 3 dog images when given 3 urls', () => {
    const tree = renderer
      .create(<FavoritesList {...baseProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the UI as expected when given no urls', () => {
    const tree = renderer
      .create(<FavoritesList {...baseProps} favorites={[]}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });


  it('does not break with no data', () => {
    const wrapper = shallow(<FavoritesList {...baseProps} favorites={[]} />)
    expect(wrapper.find('Styled(img)')).toHaveLength(0)
  })

})