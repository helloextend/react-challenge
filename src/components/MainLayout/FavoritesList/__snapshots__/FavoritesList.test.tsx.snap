// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesList component renders 3 dog images when given 3 urls 1`] = `
Array [
  <h1
    className="css-ag576t"
  >
    <img
      alt="red heart icon"
      className="css-19vi65q"
      onClick={[Function]}
      src="test-file-stub"
      style={
        Object {
          "marginRight": "30px",
        }
      }
    />
    Favorites
  </h1>,
  <div
    className="css-1740rzu"
  >
    <div
      className="css-79elbk"
    >
      <img
        alt="dog"
        className="css-aynx1p"
        src="https://images.dog.ceo/breeds/hound-afghan/n02088094_1003.jpg"
      />
      <img
        alt="white heart icon"
        className="css-19vi65q"
        onClick={[Function]}
        src="test-file-stub"
        style={
          Object {
            "bottom": "34px",
            "position": "absolute",
            "right": "10px",
          }
        }
      />
    </div>
    <div
      className="css-79elbk"
    >
      <img
        alt="dog"
        className="css-aynx1p"
        src="https://images.dog.ceo/breeds/hound-afghan/n02088094_1007.jpg"
      />
      <img
        alt="white heart icon"
        className="css-19vi65q"
        onClick={[Function]}
        src="test-file-stub"
        style={
          Object {
            "bottom": "34px",
            "position": "absolute",
            "right": "10px",
          }
        }
      />
    </div>
    <div
      className="css-79elbk"
    >
      <img
        alt="dog"
        className="css-aynx1p"
        src="https://images.dog.ceo/breeds/hound-afghan/n02088094_1023.jpg"
      />
      <img
        alt="white heart icon"
        className="css-19vi65q"
        onClick={[Function]}
        src="test-file-stub"
        style={
          Object {
            "bottom": "34px",
            "position": "absolute",
            "right": "10px",
          }
        }
      />
    </div>
  </div>,
]
`;

exports[`FavoritesList component renders the UI as expected when given no urls 1`] = `
Array [
  <h1
    className="css-ag576t"
  >
    <img
      alt="red heart icon"
      className="css-19vi65q"
      onClick={[Function]}
      src="test-file-stub"
      style={
        Object {
          "marginRight": "30px",
        }
      }
    />
    Favorites
  </h1>,
  <div
    className="css-1740rzu"
  />,
]
`;
